unitary/examples/quantum_chinese_chess/move.py

```diff
@@ -217,3 +220,418 @@ def effect(self, *objects) -> Iterator[cirq.Operation]:
         # Pass the classical properties of the source pieces to the target piece.
         target_0.reset(source_0)
         return iter(())
+
+
+class Slide(QuantumEffect):
+    """Slide from source_0 to target_0, with quantum_path_pieces_0 being the quantum pieces
+    along the path. The accepted move_variant includes
+    - CAPTURE
+    - EXCLUDED
+    - BASIC
+
+    Types of pieces that could make Slide moves are: CANNON, ROOK, HORSE, and ELEPHANT.
+    """
+
+    def __init__(
+        self,
+        quantum_path_pieces_0: List[str],
+        move_variant: MoveVariant,
+    ):
+        self.quantum_path_pieces_0 = quantum_path_pieces_0
+        self.move_variant = move_variant
+
+    def num_dimension(self) -> Optional[int]:
+        return 2
+
+    def num_objects(self) -> Optional[int]:
+        return 2
+
+    def effect(self, *objects) -> Iterator[cirq.Operation]:
+        source_0, target_0 = objects
```

Comment:
this might be error-prone, the method signature says any number of objects, but this unpacking will break if not exactly two.

Reply:
Thanks Doug. QuantumEffect._verify_objects() returns the following error message now.

```
    def _verify_objects(self, *objects: "QuantumObject"):
        if self.num_objects() is not None and len(objects) != self.num_objects():
            raise ValueError(f"Cannot apply effect to {len(objects)} qubits.")
...
```